### PR TITLE
Handle elements that aren't mounted

### LINKED ISSUE #7: Track visibility of conditionally-rendered component
We're composing this hook as part of our analytics framework and ran into an interesting case where the component we're trying to monitor the visibility of is conditionally rendered based on the loading state of some data. Specifically, the component that the `ref` returned by this hook gets attached to isn't mounted after the first render. This causes problems with this hook, as it assumes that the `ref` will have been attached to a DOM node after the first render: https://github.com/zeusdeux/use-is-in-viewport/blob/ec47313df2529a6f273e0813b202a7af1b400cee/src/index.ts#L48-58.

I believe this could be fixed by modifying the `useEffect()` call in this hook to the following:

```diff
  useEffect(() => {
+   if (childRef.current) {
      return observeElementInViewport(
        childRef.current,
        () => setIsInViewport(true),
        () => setIsInViewport(false),
        {
          ...restOpts,
          viewport: parentRef.current
        }
      )
+   }
- }, [childRef, restOpts, parentRef])
+ }, [childRef.current, restOpts, parentRef])
```

I'd be happy to put up a PR for this, but wanted to get feedback from a maintainer before doing so. Let me know if you see any problems with this approach!

Thanks for building this awesome hook 🙏

### PR DESCRIPTION
**Pull request by nwalters512**
Fixes #7 by checking if the element ref has a value before trying to observe the element's visibility. Adds two tests:

* Test mounting/unmounting the observed element
* Test mounting/unmounting the viewport element

I also refactored the test application slightly to make it easier to add tests in the future.

Some notes about the experience of working in this repo:

* It'd be great to have Prettier as the formatter for all files - having to manually think about formatting was a pain!
* The example app should adopt TypeScript to take advantage of types and TSLint - working without a linter was less than ideal.

### CODE REVIEW

**Comment by zeusdeux** [comment]
Do you mind ignoring and removing `cra/package-lock.json` from the diff please?

File: examples/cra/src/styles.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ export const app = css`
     position: fixed;
     margin-top: 51px;
   }
+  & > p {
+    position: fixed;
+    margin-top: 90px;
```
Comment (zeusdeux) [inline examples/cra/src/styles.js]:
```suggestion
    margin-top: 120px;
```
for better alignment

**Comment by nwalters512** [comment]
Made the recommended changes, thanks for reviewing!

> Do you mean in the CRA example?

Yep. I can just run Prettier on it with the editor integration, but it would be great it `npm run prettier` applied to the CRA example as well.

I also noticed that CI doesn't check for valid Prettier formatting, it would be sweet if that could fail the build.

**Comment by zeusdeux** [comment]
@nwalters512 Things should be formatted when you commit them. I see that `*.js` is missing in the lint-staged config hence the lack of it. Nevertheless, I'll think of a way to have prettier for both.

**Comment by zeusdeux** [comment]
Landed in v1.0.8 :)

**Pull request merged**